autokeras/auto/auto_model.py

```diff
@@ -214,9 +152,107 @@ def _add_node(self, input_node):
         if input_node not in self._node_to_id:
             self._node_to_id[input_node] = len(self._node_to_id)
 
+    def fit(self,
+            x=None,
+            y=None,
+            validation_data=None,
+            **kwargs):
+        # Initialize HyperGraph model
+        x = layer_utils.format_inputs(x, 'train_x')
+        y = layer_utils.format_inputs(y, 'train_y')
+
+        # TODO: Set the shapes only if they are not provided by the user when
+        #  initiating the HyperHead or Block.
+        for x_input, input_node in zip(x, self.inputs):
+            input_node.shape = x_input.shape[1:]
+        for y_input, output_node in zip(y, self.outputs):
+            if len(y_input.shape) == 1:
+                y_input = np.reshape(y_input, y_input.shape + (1,))
+            output_node.shape = y_input.shape[1:]
+            output_node.in_hypermodels[0].output_shape = output_node.shape
+
+        # Prepare the dataset
+        if validation_data is None:
+            (x, y), (x_val, y_val) = layer_utils.split_train_to_valid(x, y)
+            validation_data = x_val, y_val
+
+        self.tuner = kerastuner.RandomSearch(
+            hypermodel=self,
+            objective='val_loss',
+            max_trials=self.trials,
+            directory=self.directory)
+
+        # TODO: allow early stop if epochs is not specified.
+        self.tuner.search(x=x,
+                          y=y,
+                          validation_data=validation_data,
+                          **kwargs)
+
+    def predict(self, x, **kwargs):
+        """Predict the output for a given testing data. """
+        return self.tuner.get_best_models(1)[0].predict(x, **kwargs)
```

Comment:
I've always been cautious in indexing without prechecking the boundary, maybe it is taken cared already in keras tuner.